.github/workflows/python.yml

```diff
@@ -0,0 +1,42 @@
+name: Python CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - task: Lint with flake8
+            run: flake8 pii_recognition
+          - task: Type check
+            run: mypy --config-file pyproject.toml pii_recognition
+          - task: Unit tests
+            run: pytest pii_recognition
+          - task: Integration tests
+            run: pytest tests 
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+      - name: Get full python version
+        id: full-python-version
+        run: |
+          echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info[:3]))")
+      - name: Cache .cache dir
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache
+          key: ${{ runner.os }}-${{ steps.full-python-version.outputs.version }}-${{ hashFiles('poetry.lock') }}
```

Comment:
The cache key includes the Python version, otherwise bumping Python versions would cause re-installation of dependencies.

Reply:
No it wouldn't @gabechu  as the python version is in the poetry lock so it would invalidate the cache. There's no real need to add this I dont think

Reply:
I found that both caching and installation were triggered when change Python version `3.7.7` to `3.7` (Github fetched `3.7.9`) with `{{ runner.os }}-cache-dir-${{ hashFiles('poetry.lock') }}`. I am not sure what caused that, just suspecting maybe the version stuff.

![image](https://user-images.githubusercontent.com/60019811/97392660-4a866600-191d-11eb-8290-0a7212370265.png)

Reply:
Also, does `poetry.lock` captures which Python we've installed? A quick search shows Python version is defined in `pyproject.toml` but not in `poetry.lock` (probably I missed it).